### PR TITLE
[SQOW-172] Tech debt |  Update Searchbar API

### PR DESCRIPTION
**Pull request by jmorales-hu**
## Summary
- Updated the API of `SearchController` to export a component instead. 

## Screenshots, GIFs or Videos
N/A

## Jira Card
https://humand.atlassian.net/browse/SQOW-172

### CODE REVIEW

File: src/hooks/useClientPagination.tsx [hunk 1]
```diff
@@ -81,15 +81,33 @@ const useTableSorting = <T extends unknown>(defaultOrderBy: string, defaultOrder
...
+  limitOptions?: number[]
+} 
+
+// items: T[],
+// queriedKeys: (keyof T)[],
+// placeholder: string,
+// labelRowsPerPage: string,
+// defaultOrderBy = 'position',
+// defaultOrder: TableSortLabelProps['direction'] = 'asc',
+// limitOptions = [10, 20, 30]
```
Comment (gsequeira-hu) [inline src/hooks/useClientPagination.tsx:99]: borraria todos estos comments.

File: src/components/pagination/SearchBarController.tsx [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
+import {FormValues as ServerPaginationFormValues} from '../../hooks/useServerPagination'
 
-export type SearchBarControllerProps<T extends FieldValues> = {
-  control: Control<T>;
-  setValue: (name: keyof T, value: T[keyof T]) => void;
-  placeholder?: string;
-  isSurveys?: boolean;
-  isPeopleExperience?: boolean;
-};
+// { m: isSurveys ? 0 : 1, ...(isPeopleExperience && { flex: 1 }) }
```
Comment (gsequeira-hu) [inline src/components/pagination/SearchBarController.tsx:15]: este comment tmb lo borraria.

File: lib/hooks/useClientPagination.d.ts [hunk 3]
```diff
@@ -1,13 +1,22 @@
+/// <reference types="react" />
```
Comment (gsequeira-hu) [inline lib/hooks/useClientPagination.d.ts:1]: este comment hace falta??

Comment (jmorales-hu) [reply lib/hooks/useClientPagination.d.ts:1]: Esto se agrega automáticamente

**Review by gsequeira-hu — APPROVED** [review]

**Review by marianozicavo — APPROVED** [review]

**Pull request merged**